src/main/java/seedu/duke/ui/Formatter.java

```diff
@@ -56,9 +58,45 @@ public static String formatNote(String header, Note note) {
         return encloseTopAndBottom(formattedString);
     }
 
-    public static String formatTimetable(String header, Timetable timetable) {
-        String formattedString = "";
-        return formattedString;
+    /**
+     * Takes an array list of events and converts it to a formatted, non-indexed string for output.
+     *
+     * @param header Success message.
+     * @param year Year of timetable
+     * @param month Month of timetable
+     * @param timetable Events to be printed
+     * @return Formatted string of non-indexed events in timetable
+     */
+    public static String formatTimetable(String header, int year, int month,
+                                         HashMap<Month, HashMap<Integer, ArrayList<Event>>> timetable) {
+        String formattedString;
+        if (month != 0) {
+            formattedString = generatesHeader(header + String.format(" %d-%d", year, month));
+        } else {
+            formattedString = generatesHeader(header + " " + year);
+        }
+        ArrayList<String> unformattedEvents = Event.yearCalendarToString(timetable);
+        for (String event : unformattedEvents) {
+            formattedString = formattedString.concat(encloseRow(event));
+        }
+
+        return encloseTopAndBottom(formattedString);
+    }
+
+    /**
+     * Takes an array list of events and converts it to a formatted, indexed string for output.
+     *
+     * @param header Success message.
+     * @param events Events to be printed
+     * @return Formatted string of indexed events in timetable
+     */
+    public static String formatTimetable(String header, ArrayList<Event> events) {
+        String formattedString = generatesHeader(header);
+        ArrayList<String> eventStrings = Event.calendarToString(events);
+        for (String event : eventStrings) {
+            formattedString = formattedString.concat(encloseRow(event));
+        }
```

Comment:
In terms of reusability:
for (Event event: events) {
formattedString = formattedString.concat(formatEvent(event));
}

Reply:
Done from earlier PR

Reply:
Will resolve convo when confirmed after merging.